hubgrep_indexer/lib/state_manager/abstract_state_manager.py

```diff
@@ -166,7 +166,7 @@ def get_timed_out_block(self, hoster_prefix: str, timestamp_now=None) -> Union[B
 
         for block in self.get_blocks_list(hoster_prefix):
             age = timestamp_now - block.attempts_at[-1]
-            if not block.is_dead() and age > self.block_timeout:
+            if age > self.block_timeout:
```

Comment:
if we dont deliver dead blocks, they will never be picked up for deletion in add_repos

Reply:
todo: can we check for dead blocks (if its the last block in the run) here, and run a hoster-is-finished scenario if needed?

Reply:
@geoheelias 
actually this wouldnt be a trivial task, as we would have to (at least) split up the `host_state_helper.resolve_state()` function, and i dont want to do that. (and im not even sure if we could do that, as we dont have the block, which would be needed to test on `has_reached_end()`...?)

id say we leave this as it is: the worst thing that will happen is, that we try to crawl a "dead" block one more time, but that shouldnt happen, as we clean up dead blocks in resolve_state. its an edge case, in which a crawler tries to get results one time more than we planned.